engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/NonContainerCascadingMetaData.java

```diff
@@ -31,6 +51,22 @@
 	private static final NonContainerCascadingMetaData CASCADING_WITHOUT_GROUP_CONVERSIONS = new NonContainerCascadingMetaData( true,
 			GroupConversionHelper.of( Collections.emptyMap() ) );
 
+	private static final Set<ValueExtractorDescriptor> LIST_VALUE_EXTRACTOR_DESCRIPTOR_SET = Collections.singleton( ListValueExtractor.DESCRIPTOR );
+	private static final Set<ValueExtractorDescriptor> MAP_VALUE_EXTRACTOR_DESCRIPTOR_SET = Collections.singleton( MapValueExtractor.DESCRIPTOR );
+	private static final Set<ValueExtractorDescriptor> ITERABLE_VALUE_EXTRACTOR_DESCRIPTOR_SET = Collections.singleton( IterableValueExtractor.DESCRIPTOR );
+	private static final Set<ValueExtractorDescriptor> OPTIONAL_VALUE_EXTRACTOR_DESCRIPTOR_SET = Collections.singleton( OptionalValueExtractor.DESCRIPTOR );
+	private static final Set<ValueExtractorDescriptor> ARRAY_VALUE_EXTRACTOR_DESCRIPTOR_SET = CollectionHelper.asSet(
+			BooleanArrayValueExtractor.DESCRIPTOR,
+			ByteArrayValueExtractor.DESCRIPTOR,
+			CharArrayValueExtractor.DESCRIPTOR,
+			DoubleArrayValueExtractor.DESCRIPTOR,
+			FloatArrayValueExtractor.DESCRIPTOR,
+			IntArrayValueExtractor.DESCRIPTOR,
+			LongArrayValueExtractor.DESCRIPTOR,
+			ObjectArrayValueExtractor.DESCRIPTOR,
+			ShortArrayValueExtractor.DESCRIPTOR
```

Comment:
What if place these collections in `ValueExtractorManager` and either make them public or provide  "static getters" for them. Then there will be no need in exposing extractor implementations out of the package. Or is the manager just not the right place for them ?

Reply:
Yes, sure, we can discuss this.

I thought about putting them in the manager but I didn't find that compelling in the end.

Reply:
I just saw those visibility changes from default to public so I thought I'll ask (to better understand the preferences with these visibility things :) ). If it's OK for those extractors to be exposed then I guess everything is fine :)

I think this is a bit related to that question about those cycles between packages and how tightly the packages are coupled with other packages and things like that...

Reply:
I'm thinking about introducing a LegacyCollectionValueExtractors class in the said package. But not sure it has much value.

The idea is to reduce the visibility as much as possible but we have a lot of public stuff that are not really public and are just public because we isolated code in a package for clarity.

I admit though it's a bit weird to have part of the extractors public and part of them package protected.

Reply:
@marko-bekhta OK, did that and made the visibility of the value extractors consistent. Some of them were already uselessly public.

Reply:
nice, and thanks for the explanation about the visibility :)